documentation/modules/exploit/linux/local/ufo_privilege_escalation.md

```diff
@@ -0,0 +1,166 @@
+## Vulnerable Application
+
+This module attempts to gain root privileges on Linux systems by abusing UDP Fragmentation Offload (UFO).
+
+This module has been tested successfully on:
+
+  * Ubuntu 14.04.5 4.4.0-31-generic x64 Desktop
+  
+The following kernels are vulnerable, but untested on Ubuntu
```

Comment:
A huge number of kernels are affected, from somewhere around `3.13` to somewhere around `4.11`.

[Allegedly](https://ricklarabee.blogspot.com.au/2017/12/adapting-poc-for-cve-2017-1000112-to.html) the issue is also exploitable on other operating systems, such as CentOS.

The exploit requirement for user namespaces puts a lower bound, regardless of vulnerable kernel version, at `kernel_version < Gem::Version.new('3.12')`.